bundles/org.openhab.binding.shelly/src/main/resources/ESH-INF/thing/sensor.xml

```diff
@@ -0,0 +1,240 @@
+            <channel id="tempUnit"    typeId="sensorTempUnit" />
+            <channel id="humidity"    typeId="sensorHumidity" />
+            <channel id="last_update" typeId="lastUpdate" />
+            <channel id="event"       typeId="eventTrigger" />
+        </channels>
+    </channel-group-type>
+    
+    <channel-group-type id="floodSensor">
+       <label>Sensor Data</label>
+       <description>Data from the Flood Sensor</description>
+        <channels>
+            <channel id="temperature" typeId="sensorTemp" />
+            <channel id="tempUnit"    typeId="sensorTempUnit" />
+            <channel id="flood"       typeId="sensorFlood" />
+            <channel id="rainMode"    typeId="sensorRainMode" />
+            <channel id="last_update" typeId="lastUpdate" />
+            <channel id="event"       typeId="eventTrigger" />
+        </channels>
+   </channel-group-type>
+
+   <channel-group-type id="senseSensors">
+       <label>Sensors</label>
+       <description>Data from the Sense sensors</description>
+        <channels>
+            <channel id="temperature" typeId="sensorTemp" />
+            <channel id="tempUnit"    typeId="sensorTempUnit" />
+            <channel id="humidity"    typeId="sensorHumidity" />
+            <channel id="lux"         typeId="sensorLux" />
+            <channel id="motion"      typeId="sensorMotion" />
+            <channel id="charger"     typeId="sensorCharger" />
+            <channel id="last_update" typeId="lastUpdate" />
+            <channel id="event"       typeId="eventTrigger" />
+        </channels>
+    </channel-group-type>
+
+   <channel-group-type id="batteryStatus">
+       <label>Battery Status</label>
+       <description>Status of the device's battery</description>
+        <channels>
+            <channel id="batteryLevel" typeId="batLevel" />
+            <channel id="lowBattery"   typeId="batAlert" />
+            <channel id="voltage"      typeId="batVoltage" />
+        </channels>
+    </channel-group-type>
+   <channel-group-type id="shortBatteryStatus">
+       <label>Battery Status</label>
+       <description>Status of the device's battery</description>
+        <channels>
+            <channel id="batteryLevel" typeId="batLevel" />
+            <channel id="lowBattery"   typeId="batAlert" />
+        </channels>
+    </channel-group-type>
+
+   <channel-group-type id="senseControl">
+       <label>Sense Control</label>
+       <description>Sense Settings</description>
+        <channels>
+            <channel id="key" typeId="senseKey" />
+        </channels>
+    </channel-group-type>
+
+    <channel-type id="sensorTemp">
+        <item-type>Number</item-type>
+        <label>Temperature</label>
+        <description>Temperature from the sensor</description>
+        <state readOnly="true" pattern="%.1f" >
+        </state>
+    </channel-type>
+    <channel-type id="sensorTempUnit">
+        <item-type>String</item-type>
```

Comment:
Please remove channel, it's a device config

Reply:
Why should we require the user to set this config in the Shelly App AND the binding config.
tempUnit is read from the device status and can be used by ruiles etc. to find out of the temperature value is in C or F. So if the user changes the device setting in the Shelly App or Web Ui the binding automatically reports this value without another config change.

Reply:
see above